File: backend/app/schemas/character.py
from datetime import datetime

from pydantic import BaseModel


class CharacterCreate(BaseModel):
    name: str
    name_reading: str | None = None
    age: int | None = None
    occupation: str | None = None
    personality_type: str | None = None
    personality_traits: list[str] | None = None
    speaking_style: str | None = None
    catchphrase: str | None = None
    interests: list[str] | None = None
    backstory: str | None = None
    system_prompt: str
    greeting_message: str | None = None
    sample_dialogues: list[dict] | None = None
    avatar_color: str | None = None
    category: str = "healing"


class CharacterResponse(BaseModel):
    id: str
    name: str
    name_reading: str | None
    age: int | None
    occupation: str | None
    personality_type: str | None
    personality_traits: list | None
    speaking_style: str | None
    catchphrase: str | None
    interests: list | None
    backstory: str | None
    system_prompt: str
    greeting_message: str | None
    sample_dialogues: list | None
    avatar_color: str | None
    category: str
    is_preset: bool
    creator_id: str | None
    usage_count: int
    created_at: datetime

    model_config = {"from_attributes": True}


class AgentFromCharacterResponse(BaseModel):
    agent_id: str
    character_id: str
    character_name: str

File: ios/Clawzy/Models/Character.swift
import Foundation

struct CharacterTemplate: Codable, Identifiable {
    let id: String
    let name: String
    let nameReading: String?
    let age: Int?
    let occupation: String?
    let personalityType: String?
    let personalityTraits: [String]?
    let speakingStyle: String?
    let catchphrase: String?
    let interests: [String]?
    let backstory: String?
    let systemPrompt: String
    let greetingMessage: String?
    let sampleDialogues: [SampleDialogue]?
    let avatarColor: String?
    let category: String
    let isPreset: Bool
    let creatorId: String?
    let usageCount: Int
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, name, age, occupation, backstory, category, interests, catchphrase
        case nameReading = "name_reading"
        case personalityType = "personality_type"
        case personalityTraits = "personality_traits"
        case speakingStyle = "speaking_style"
        case systemPrompt = "system_prompt"
        case greetingMessage = "greeting_message"
        case sampleDialogues = "sample_dialogues"
        case avatarColor = "avatar_color"
        case isPreset = "is_preset"
        case creatorId = "creator_id"
        case usageCount = "usage_count"
        case createdAt = "created_at"
    }
}

struct SampleDialogue: Codable {
    let user: String
    let char: String
}

struct UseCharacterResponse: Codable {
    let agentId: String
    let characterId: String
    let characterName: String

    enum CodingKeys: String, CodingKey {
        case agentId = "agent_id"
        case characterId = "character_id"
        case characterName = "character_name"
    }
}
